src/client/activation/analysis.ts

```
@@ -96,7 +88,7 @@ export class AnalysisExtensionActivator implements IExtensionActivator {
         return false;
     }
 
-    private async tryStartLanguageClient(context: ExtensionContext, lc: LanguageClient): Promise<Error> {
+    private async tryStartLanguageClient(context: ExtensionContext, lc: LanguageClient): Promise<Error | undefined> {

```

oops, not sure how I missed this. Why are we returning a promise?
Shouldn't we return an error as part of the rejection. I.e. change this to `Promise<void>` and errors will be captured in `catch`?

---

Yeah, you are right. Returning exception is questionable pattern.